File: src/lib/Entry.cpp
#include "Entry.h"

// empty constructor
Entry::Entry() {}

// constructor from string
Entry::Entry(string s) {
    raw_string = s;
    vector<string> split = spilt_string(s, ',');

    // if times are valid add them else throw exception
    if(valid_time(split[0]) && valid_time(split[3])) {
        departure_time = string_to_time(split[0]);
        arrival_time = string_to_time(split[3]);
    }
    else throw invalid_argument("Entry is not on the same day");

    id = split[1];

    string temp = split[2];
    TOLOWER(temp);
    departure_location = temp;

    temp = split[4];
    TOLOWER(temp);
    arrival_location = temp;
}

File: src/lib/Entry.py
from lib.Helper import string_to_time, valid_time

class Entry:
    def __init__(self, rawstring):
        self.raw_string = rawstring.strip("\n")
        s = self.raw_string.split(",")

        self.id = s[1]
        self.departure_location = s[2].lower()
        self.arrival_location = s[4].lower()

        if valid_time(s[0]) and valid_time(s[3]):
            self.departure_time = string_to_time(s[0])
            self.arrival_time = string_to_time(s[3])
        else:
            raise ValueError("Time not in same day")


    def __repr__(self):
        return self.raw_string